code/game/machinery/computer/depot.dm

```diff
@@ -0,0 +1,481 @@
+
+/obj/machinery/computer/syndicate_depot/syndiecomms/secondary(var/mob/user, var/subcommand)
+	if(..())
+		return
+	if(has_security_lockout(user))
+		return
+	if(depotarea)
+		if(depotarea.local_alarm || depotarea.called_backup || depotarea.used_self_destruct)
+			to_chat(user, "<span class='warning'>Visitor sign-in is not possible while the depot is on security alert.</span>")
+		else if(depotarea.on_peaceful)
+			if(subcommand == DEPOT_VISITOR_END)
+				if(check_rights(R_ADMIN, 0, user))
+					depotarea.peaceful_mode(FALSE, TRUE)
+			else if (subcommand == DEPOT_VISITOR_ADD)
+				if(user.mind && user.mind.special_role == SPECIAL_ROLE_TRAITOR)
+					if(user in depotarea.peaceful_visitors)
+						to_chat(user, "<span class='warning'>[user] is already signed in as a visiting agent.</span>")
+					else
+						grant_syndie_faction(user)
+				else
+					to_chat(user, "<span class='warning'>Only verified agents of the Syndicate may sign in as visitors. Everyone else will be shot on sight.</span>")
+		else if(subcommand == DEPOT_VISITOR_START)
+			if(depotarea.something_looted)
+				to_chat(user, "<span class='warning'>Visitor sign-in is not possible after supplies have been taken from the depot.</span>")
+			else if(user.mind && user.mind.special_role == SPECIAL_ROLE_TRAITOR)
+				grant_syndie_faction(user)
+				depotarea.peaceful_mode(TRUE, TRUE)
+			else
+				to_chat(user, "<span class='warning'>Only verified agents of the Syndicate may sign in as visitors. Everyone else will be shot on sight.</span>")
+		else
+			to_chat(user, "<span class='warning'>Unrecognized subcommand: [subcommand]</span>")
+	else
+		to_chat(user, "<span class='warning'>ERROR: [src] is unable to uplink to depot network.</span>")
+	updateUsrDialog()
+
+/obj/machinery/computer/syndicate_depot/syndiecomms/proc/grant_syndie_faction(var/mob/user)
+	user.faction += "syndicate"
+	depotarea.peaceful_visitors += user
+	to_chat(user, {"<BR><span class='userdanger'>Welcome, Agent.</span>
+		<span class='warning'>You are now signed-in as a depot visitor.
+		Any other agents with you MUST sign in themselves.
+		You may explore all rooms here, except for bolted ones.
+		Your agent ID will give you access to most doors and computers.
+		Standard Syndicate regulations apply to your visit.
+		This means if ANY of you attack facility staff, break into anything, sabotage the facility, or bring non-agents here, then ALL of you will be summarily executed.
+		Enjoy your stay.</span>
+	"})
+
+/obj/machinery/computer/syndicate_depot/syndiecomms/power_change()
+	. = ..()
+	if(!security_lockout && (stat & NOPOWER))
+		security_lockout = TRUE
+		raise_alert("[src] lost power.")
+
+
+// Syndicate teleporter control, used to manage incoming/outgoing teleports
+
+/obj/machinery/computer/syndicate_depot/teleporter
+	name = "syndicate teleporter console"
+	icon_screen = "telesci"
+	icon_keyboard = "teleport_key"
+	var/obj/machinery/bluespace_beacon/syndicate/mybeacon
+	var/turf/tele_target
```

Comment:
same.

Reply:
What's the best way to do that? I can't really create a subtype of every possible turf with a backlink.